apps/src/applab/appStorage.js

```diff
@@ -33,6 +33,10 @@ var handleGetKeyValue = function(onSuccess, onError) {
     onSuccess(undefined);
     return;
   }
+  if (this.status === 429) {
```

Comment:
This code snippet is duplicated in too many places; please use an onErrorStatus(message, status) helper function instead. For example   

 if (this.status < 200 || this.status >= 300) {
      onErrorStatus( 'Error writing value', this.status);
      return;
}
...
function onErrorStatus(message, status) {
   onError(message + ':' + getStatusDescription(status));
}

function getStatusDescription(status) {
  if (status === 429) { 
    return 'rate limit exceeded';
 } else {
     return 'unexpected http status ' + status);
 }
}